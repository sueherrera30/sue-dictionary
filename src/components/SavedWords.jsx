import { useNavigate } from "react-router-dom";

import styles from './styles/savedWords.module.css';
import hearts from '../assets/favs.png';
const SavedWords = () => {
    let navigate = useNavigate();
    return (
        <div className={styles.savedContainer} onClick={() => navigate('/favorites')}>
            <div className={styles.imageContainer}>
                <img src={hearts} alt="main" className={styles.image}/>
            </div>
            <p className={styles.text}>MY FAV WORDS </p>
        </div>
    )
};

export default SavedWords;