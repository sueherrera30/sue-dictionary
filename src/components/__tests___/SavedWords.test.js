import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import {  MemoryRouter } from 'react-router-dom';

import SavedWords from '../SavedWords';

const mockedUsedNavigate = jest.fn();

afterEach(() => {
    cleanup();
});

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
  }));

test('<SavedWords />', () => {
    render(<MemoryRouter><SavedWords /></MemoryRouter>, );
    expect(screen.container).toMatchSnapshot();
    expect(screen.getByText('MY FAV WORDS')).toBeTruthy();
});

test('Back to favorites', async () => {
    render(<MemoryRouter><SavedWords /></MemoryRouter>, );
    const favButton = screen.queryByTestId('favorites-button');
    await fireEvent.click(favButton);
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/favorites')
});