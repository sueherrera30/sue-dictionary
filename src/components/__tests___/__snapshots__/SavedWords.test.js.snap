// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavedWords /> 1`] = `undefined`;
